Stop the `config_manager` child process on exception in `chassisd`

arista-nwolfe opened the pull request:
Context is in: https://github.com/sonic-net/sonic-buildimage/issues/24775

TLDR:
`chassisd` process doesn't die when a crash occurs because the `config_manager` child process is still running.
Adding `try/finally` to cleanup the `config_manager` properly when a crash occurs to let `chassisd` die and restart.

Confirmed that if I add an `assert False` after `config_manager` is created I see the `chassisd` process die and restart as expected (previously the process would live on).

Note: When reviewing the diff select `hide whitespaces` to make the diff easier to read.

Cast Request:
- 202405

arista-nwolfe committed:
Stop the config_manager child process on exception in chassisd

https://github.com/sonic-net/sonic-buildimage/issues/24775
Signed-off-by: Nathan Wolfe <nwolfe@arista.com>

YatishSVC reviewed (approved):
lgtm, Thanks

anamehra reviewed (approved):
LGTM

arlakshm reviewed (commented):
@nwolfe, is it possible to add UT for this?

arista-nwolfe committed:
Add a Unit Test to cover the new change

UT confirms that the child thread config_manager is killed if the main
loop hits an assertion error.

Signed-off-by: Nathan Wolfe <nwolfe@arista.com>

mssonicbld commented:
/azp run

arista-nwolfe committed:
Adding missed argument for MockChassis

Signed-off-by: Nathan Wolfe <nwolfe@arista.com>

arista-nwolfe committed:
Modified test to run on supervisor slot

Signed-off-by: Nathan Wolfe <nwolfe@arista.com>
